src/App.tsx

```diff
@@ -15,81 +28,128 @@ export const goodsFromServer = [
-  return (
-    <div className="section content">
-      <div className="buttons">
-        <button
-          type="button"
-          className="button is-info is-light"
-        >
-          Sort alphabetically
-        </button>
-
-        <button
-          type="button"
-          className="button is-success is-light"
-        >
-          Sort by length
-        </button>
-
-        <button
-          type="button"
-          className="button is-warning is-light"
-        >
-          Reverse
-        </button>
-
-        <button
-          type="button"
-          className="button is-danger is-light"
-        >
-          Reset
-        </button>
-      </div>
+  toggleIsReversed(prevsTate: boolean) {
+    this.setState(() => ({ isReversed: prevsTate }));
+  }
+
+  reset() {
+    this.setState({
+      sortType: SortType.NONE, isReversed: false,
+    });
+  }
+
+  render(): React.ReactNode {
+    const reorderedGoods = getReorderedGoods(this.state.goodsList, this.state);
+
+    return (
+      <div className="section content">
+        <div className="buttons">
+          <button
+            type="button"
+            className={`button is-info ${this.state.sortType !== SortType.ALPHABET ? 'is-light' : ''}`}
+            onClick={() => {
+              this.setsortType(SortType.ALPHABET);
+            }}
+          >
+            Sort alphabetically
+          </button>
+
+          <button
+            type="button"
+            className={`button is-success ${this.state.sortType !== SortType.LENGTH ? 'is-light' : ''}`}
+            onClick={() => {
+              this.setsortType(SortType.LENGTH);
+            }}
+          >
+            Sort by length
+          </button>
+
+          <button
+            type="button"
+            className={this.state.isReversed
+              ? 'button is-warning'
+              : 'button is-success is-light'}
+            onClick={() => {
+              this.toggleIsReversed(!this.state.isReversed);
+            }}
+          >
+            Reverse
+          </button>
+
+          {this.state.isReversed === true
+          || this.state.sortType !== SortType.NONE
+            ? (
+              <button
+                type="button"
+                className="button is-danger is-light"
+                onClick={() => {
+                  this.reset();
+                }}
+              >
+                Reset
+              </button>
+            )
+            : null}
+        </div>
 
-      <ul>
         <ul>
-          <li data-cy="Good">Dumplings</li>
-          <li data-cy="Good">Carrot</li>
-          <li data-cy="Good">Eggs</li>
-          <li data-cy="Good">Ice cream</li>
-          <li data-cy="Good">Apple</li>
-          <li data-cy="Good">...</li>
+          <ul>
+            { reorderedGoods.map((goodSecond) => {
+              return (
+                <li data-cy="Good" key={goodSecond}>{goodSecond}</li>
+              );
+            })}
```

Comment:
it can be simplified
```suggestion
            {reorderedGoods.map((goodSecond) => (
              <li data-cy="Good" key={goodSecond}>{goodSecond}</li>
            ))}
```